src/components/render-form.component.js

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable array-callback-return */
+/* eslint-disable no-param-reassign */
+import { renderTemplate } from '../utils/render-template';
+import { notifyStateUpdated } from '../state/root.state';
+import { renderListItem } from './render-list-item.component';
+
+export const renderForm = (formState, onDestroy) => {
+  const editableForm = renderTemplate(`
+  <div class="card-form">
+    <div>
+      <input class="card__title card-form__title" placeholder="I am a title" id="title" value=${formState.title}>
+      <ul class="card-form__list"></ul>
+      <div class="add-item">
+        <svg class="add-item__icon" fill="rgba(51, 51, 51)" width="18" height="18">
+          <use xlink:href="#plus"></use>
+        </svg>
+        <input type="text" class="add-item__input" placeholder="List item..." />
+      </div>
+    </div>
+    <div class="card-footer--visible">
+      <svg class="card-footer__icon" fill="rgba(1, 1, 1, 1)" width="13.05px" height="15.06px">
+        <use xlink:href="#bin"></use>
+      </svg>
+    </div>
+  </div>
+`);
+
+  const cardContainer = editableForm.querySelector('.card-form');
+  const formTitle = editableForm.querySelector('.card-form__title');
+  const addFormItemIcon = editableForm.querySelector('.add-item__icon');
+  const addFormItemInput = editableForm.querySelector('.add-item__input');
+  const formListContainer = editableForm.querySelector('.card-form__list');
+  const formDeleteIcon = editableForm.querySelector('.card-footer__icon');
+
+  const itemState = formState.items;
+
+  itemState.map((el) => {
```

Comment:
What is the difference between `forEach` and `map`?

Reply:
`map` returns new array according to result provided by the function we call. forEach does not. It returns undefined. They both mutate the original array (by means of callback provided)